src/utils/fees.ts

```diff
@@ -17,11 +18,24 @@ const feePriceInUnit = (asset: Asset, feePrice: number) => {
   return isEthereumChain(asset) ? new BN(feePrice).times(1e9) : feePrice; // ETH fee price is in gwei
 };
 
-function getSendFee(asset: Asset, feePrice: number) {
+function getSendFee(asset: Asset, feePrice: number, l1FeePrice?: number, network?: Network) {
   const assetInfo = cryptoassets[asset];
-  const fee = new BN(assetInfo.sendGasLimit).times(feePriceInUnit(asset, feePrice));
-
-  return unitToCurrency(cryptoassets[getNativeAsset(asset)], fee);
+  const nativeAsset = cryptoassets[getNativeAsset(asset)];
+
+  if (l1FeePrice && isMultiLayeredChain(assetInfo.chain)) {
+    if (assetInfo.chain === ChainId.Optimism) {
```

Comment:
Do you need to check both `isMultiLayeredChain` and the chain? Since you know this will only work for optimisim

In the case that this if condition returns false, the function returns undefined but it should never
I think a better structure might be 
```
 if (assetInfo.chain === ChainId.Optimism) {
    if(!l1feePrice) { 
      throw new Error('l1FeePrice  must be specified for optimism')
    }
 }
```

This will prevent this function having the `| undefined` return type